bundle/direct/dresources/secret_scope.go

```diff
@@ -0,0 +1,92 @@
+package dresources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/cli/libs/utils"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+)
+
+type ResourceSecretScope struct {
+	client *databricks.WorkspaceClient
+}
+
+type SecretScopeConfig struct {
+	workspace.CreateScope
+}
+
+func (*ResourceSecretScope) New(client *databricks.WorkspaceClient) *ResourceSecretScope {
+	return &ResourceSecretScope{
+		client: client,
+	}
+}
+
+func (*ResourceSecretScope) PrepareState(input *resources.SecretScope) *SecretScopeConfig {
+	return &SecretScopeConfig{
+		CreateScope: workspace.CreateScope{
+			Scope:                  input.Name,
+			ScopeBackendType:       input.BackendType,
+			BackendAzureKeyvault:   input.KeyvaultMetadata,
+			InitialManagePrincipal: "",
+			ForceSendFields:        nil,
+		},
+	}
+}
+
+func (*ResourceSecretScope) RemapState(remote *workspace.SecretScope) *SecretScopeConfig {
+	return &SecretScopeConfig{
+		CreateScope: workspace.CreateScope{
+			Scope:                  remote.Name,
+			ScopeBackendType:       remote.BackendType,
+			BackendAzureKeyvault:   remote.KeyvaultMetadata,
+			InitialManagePrincipal: "",
+			ForceSendFields:        utils.FilterFields[workspace.CreateScope](remote.ForceSendFields),
+		},
+	}
+}
+
+func (r *ResourceSecretScope) DoRefresh(ctx context.Context, id string) (*workspace.SecretScope, error) {
```

Comment:
Why do we need to implement DoRefresh here?

Reply:
+1, DoRefresh() is required but it seems that it could be a no-op, given that the scopes are immutable?

Reply:
How would we do remote diff detection without a DoRefresh method? Or even that the scope was accidentally deleted? Scopes do not have a GET by ID method.

Reply:
Good point about presence. We need to know if resource still exists.